### PR TITLE
Enable nullability in some PropertyGridView members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some PropertyGridView members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10009)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 1]
```diff
@@ -2297,26 +2297,26 @@ protected override void OnHandleDestroyed(EventArgs e)
...
     }
 
-    private void OnListMouseUp(object sender, MouseEventArgs me)
+    private void OnListMouseUp(object? sender, MouseEventArgs me)
     {
         OnListClick(sender, me);
     }
 
-    private void OnListClick(object sender, EventArgs e)
+    private void OnListClick(object? sender, EventArgs e)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2319]: why not `private void OnListClick(object? sender, EventArgs? e)` and avoid `!` on caller?  Its privat emethod.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 2]
```diff
@@ -2405,11 +2405,11 @@ protected override void OnLostFocus(EventArgs e)
 
         if (FocusInside)
         {
-            base.OnLostFocus(e);
+            base.OnLostFocus(e!);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: I don't see `e` being null here? why `!`?

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: And also condition on line 2401.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: There are warnings without these. I guess due to `if (e is not null)` above. We had to use `!` in some similar cases in the past.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: Check `OnMouseWheel` in `ScrollBar` class for example.

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: We need to revisit those cases with a practical scenario here.

If `e` can be null, `OnLostFocus` should be annotated so. Reading this code as is doesn't sound right. If base isn't expecting null, but override method expects null and doesn't annotate.

Can you test the scenario and see if there is ever a case where `e` could be null here?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: Check https://github.com/dotnet/winforms/pull/7491/files#r928648176 for a related discussion. I could add the same comment here.

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:2408]: Thank you for pointing me to this. Given it was already discussed and finalized, lets go with comment (easy to revisit if we decided to make breaking change).

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**